CMakeLists.txt

```diff
@@ -155,6 +167,26 @@ if (NOT "${NSYNC_LANGUAGE}X" STREQUAL "c++11X")
 	elseif ("${CMAKE_SYSTEM_NAME}X" STREQUAL "OpenBSDX")
 		include_directories ("${PROJECT_SOURCE_DIR}/platform/openbsd")
 		set (NSYNC_POSIX ON)
+      elseif ( "${CMAKE_SYSTEM_NAME}X" STREQUAL "AndroidX")
+
+        message("ANDROID!!!")
+        
+      if ("${ANDROID_ARCH}X" STREQUAL "armv7-aX")
+		include_directories ("${PROJECT_SOURCE_DIR}/platform/arm")
+		set (NSYNC_POSIX ON)        
+      elseif ("${ANDROID_ARCH}X" STREQUAL "armv7-a-hardX")
```

Comment:
`...X` trick is not saving us from anything, it's better to use `string(COMPARE EQUAL ...)` or update requirement to CMake 3.1+.

Reply:
I made use of `string(COMARE EQUAL ...)` for the localized android specific patches.